### ISSUE TITLE
[espanso-detect] Espanso Sync Tool window not rendering correctly on wayland

### ISSUE THREAD

**Issue by morgan-dgk** [federico-terzi/espanso#1776]
**Describe the bug**
The window created by espanso-detect fails to render correctly on Wayland. The window is created but the Espanso logo and text is not shown. Instead, the window is filled with a rainbow gradient. 

**To Reproduce**
1. Run espanso worker or espanso service start

**Expected behavior**
The espanso sync tool window should display the logo and text correctly. 

**Screenshots**

![image](https://github.com/espanso/espanso/assets/13283822/12c1dcbd-2753-4520-8d7c-ed45c747ae81)

**Logs**
Relevant logs are from Wayland compositor. 

```
[3375819.298] wl_callback@3.done(89687)
[3375819.312]  -> wl_registry@2.bind(9, "zwp_primary_selection_device_manager_v1", 1, new id [unknown]@3)
[3375819.340]  -> zwp_primary_selection_device_manager_v1@3.get_device(new id zwp_primary_selection_device_v1@14, wl_seat@8)
[3375819.380]  -> wl_compositor@5.create_surface(new id wl_surface@15)
[3375819.395]  -> wl_registry@2.bind(13, "xdg_wm_base", 2, new id [unknown]@16)
[3375819.437]  -> wl_shm@4.create_pool(new id wl_shm_pool@17, fd 20, 4096)
[3375819.477]  -> xdg_wm_base@16.get_xdg_surface(new id xdg_surface@18, wl_surface@15)
[3375819.490]  -> xdg_surface@18.get_toplevel(new id xdg_toplevel@19)
[3375819.499]  -> wl_surface@15.commit()
[3375819.511]  -> xdg_toplevel@19.set_min_size(2, 1)
[3375819.519]  -> xdg_surface@18.set_window_geometry(0, 0, 1, 1)
[3375819.530]  -> wl_compositor@5.create_surface(new id wl_surface@20)
[3375819.539]  -> wl_seat@8.get_pointer(new id wl_pointer@21)
[3375819.554]  -> zxdg_decoration_manager_v1@9.get_toplevel_decoration(new id zxdg_toplevel_decoration_v1@22, xdg_toplevel@19)
[3375819.568]  -> xdg_toplevel@19.set_title("Espanso Sync Tool")
[3375819.587]  -> wl_shm@4.create_pool(new id wl_shm_pool@23, fd 22, 4096)
[3375819.605]  -> wl_seat@8.get_keyboard(new id wl_keyboard@24)

```

**Setup information**
- OS: Arch Linux 6.6.4 / Hyprland v.0.33.1
- Version:  2.2.0 built from source with latest fix for sctk from dev branch.

**Comment by morgan-dgk** [comment]
Update: the wayland surface created is also unresponsive and cannot be closed via normal means (espanso process must be killed).

**Comment by xircon** [comment]
No help,  but I am also seeing this,  I have been moving the window to desktop 10, as a work around.

**Comment by jdambron** [comment]
I am experiencing the same issue.

**Comment by jpetazzo** [comment]
Same, and here is a bit more information:

- this happens on machines with Intel GPU as well as NVIDIA
  (I initially thought this was an NVIDIA issue since NVIDIA's Wayland
  support is known to be... suboptimal 😅)

- it looks like the regression happened between 2.2.0 and 2.2.1 
  (I have the rainbow window on 2.2.1, but not on 2.2.0)

**Comment by daniel-r-jackson** [comment]
>     * this happens on machines with Intel GPU as well as NVIDIA
>       (I initially thought this was an NVIDIA issue since NVIDIA's Wayland
>       support is known to be... suboptimal 😅)

I'm also seeing it with amdgpu.

**Comment by igorrizhyi** [comment]
same for me with amd as well

**Comment by bsamseth** [comment]
> * it looks like the regression happened between 2.2.0 and 2.2.1
>       (I have the rainbow window on 2.2.1, but not on 2.2.0)

Downgrading did _not_ change anything, same issue still. Looking at the diff between those versions that is as expected, as only Windows related code changed.

**Comment by morgan-dgk** [comment]
The regression was introduced in 2.20 due to upgrade of smithay-client-toolkit (sctk) dependency from v0.15.4 to v0.16.1 in espanso-detect on wayland. Prior to this upgrade, espanso was crashing on launch under wayland due to a mismatch between protocol versions offered by the server and those used in the client library.

I have done some digging on this and as far as I can tell there is no protocol error occuring (this would cause the server to kill the connection with client). When the xdg_toplevel is destroyed, the associated wl_surface should also be destroyed but this does not appear to be occuring.  

Bumping the sctk dependency again _may_ fix this behaviour, but there are breaking changes to the API in both 17.0 and the latest version.

**Comment by KAGEYAM4** [comment]
Are there any workaround for this?

**Comment by bentolor** [comment]
> Are there any workaround for this?

The PR #1849 contains a fix which seem to resolve the issue. No binary/direct download available. But there is a great description how to build this PR. Otherwise, downgrading might be a workaround as well. @KAGEYAM4

**Comment by nemonein** [comment]
On Archlinux, KDE Plasma6/wayland.

'Espanso Sync tool' stays on Task manager. It can't be quit.
![espanso-sync-tool](https://github.com/espanso/espanso/assets/5688288/f5868837-59cd-4c2a-a1b2-3f2b1aac2d3e)

And when the system is going to shut down, from time to time, it fails to quit espanso sync tool.

**Comment by KwisatzJim** [comment]
> On Archlinux, KDE Plasma6/wayland.
> 
> 'Espanso Sync tool' stays on Task manager. It can't be quit. ![espanso-sync-tool](https://private-user-images.githubusercontent.com/5688288/311995285-f5868837-59cd-4c2a-a1b2-3f2b1aac2d3e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAyMzI3OTAsIm5iZiI6MTcxMDIzMjQ5MCwicGF0aCI6Ii81Njg4Mjg4LzMxMTk5NTI4NS1mNTg2ODgzNy01OWNkLTRjMmEtYTFiMi0zZjJiMWFhYzJkM2UucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMTJUMDgzNDUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2Y3ZGU2MmRhOTA5OTFlZWI5YmYxMWZiMTFhMjIyNzViYzUxNTQ2NmIwM2IwZjNjY2Y4ZGE5YTUxOTQ5NGJlMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.9EyZUXCRtIKzqVVCi801kJbu2iUZV7ZWju6BkTcTHEQ)
> 
> And when the system is going to shut down, from time to time, it fails to quit espanso sync tool.

I've experienced the "fails to quit espanso syc tool" on shutdown or reboot after upgrading to Plasma 6. Didn't happen with Plasma 5

**Comment by AucaCoyan** [comment]
closed by  #1901 🚀

**Comment by semi710** [comment]
@AucaCoyan i am still facing this issue.
Is the fix available in the release branch?
![image](https://github.com/espanso/espanso/assets/60490474/d868896f-9a46-4eed-a819-9939e5601649)
i used the version: `2.2.1-4`

**Comment by AucaCoyan** [comment]
Nope, it's on the `dev` branch, it's not yet on production

**Comment by iamkarlson** [comment]
Do you have any guesstimate on the next release date? this window is a bit annoying

**Comment by GianniBYoung** [comment]
any word on when #1901 will hit main?

**Comment by smeech** [comment]
> any word on when #1901 will hit main?

What do you mean by "main"? #1901 has been merged into the dev (default) branch, so you will get the benefit if you compile from there. We don't have a date for the next release, however.

**Comment by GianniBYoung** [comment]
> What do you mean by "main"?

by main I meant the master branch where releases are built from.

**Comment by AucaCoyan** [comment]
Yup, what smeech said is correct. We don't have a date for the next release

**Comment by MRDGH2821** [comment]
This window prevents my device from shutting down instantly. Now I'm just waiting for new release to come which includes the fixes.

**Comment by smeech** [comment]
> by main I meant the master branch where releases are built from.

You may mean "master", but recent releases have been built from the "dev" branch. Not sure why!

**Comment by firefueled** [comment]
Building from source at dev branch fixes the issue for me on Manjaro.
It was pretty easy and fast.
I had to re-register the service with `espanso service unregister && espanso service register`

**Comment by AucaCoyan** [comment]
This fix is already merged in the latest release, v2.2.3 👌🏼
